sql/core/src/main/scala/org/apache/spark/sql/streaming/DataStreamWriter.scala

```diff
@@ -300,54 +301,53 @@ final class DataStreamWriter[T] private[sql](ds: Dataset[T]) {
         "write files of Hive data source directly.")
     }
 
-    if (source == "memory") {
-      assertNotPartitioned("memory")
+    if (source == SOURCE_NAME_TABLE) {
+      assertNotPartitioned(SOURCE_NAME_TABLE)
+
+      import df.sparkSession.sessionState.analyzer.CatalogAndIdentifier
+
+      import org.apache.spark.sql.connector.catalog.CatalogV2Implicits._
+      val CatalogAndIdentifier(catalog, identifier) = df.sparkSession.sessionState.sqlParser
+          .parseMultipartIdentifier(tableName)
+
+      // Currently we don't create a logical streaming writer node in logical plan, so cannot rely
+      // on analyzer to resolve it. Directly lookup only for temp view to provide clearer message.
+      // TODO (SPARK-27484): we should add the writing node before the plan is analyzed.
+      if (isTempView(df.sparkSession, identifier.asMultipartIdentifier)) {
```

Comment:
This is incorrect. The `identifier` is for a specific catalog. e.g.  for `cat1.t1`, the `identifier` is `t1`.  `cat1.t1` is not a temp view, but `t1` might be.

We should check temp view with the original table name.

Reply:
Please correct me if I'm missing here. The reason I pass all parts in identifier here is to cover global temp view, which uses global temp db. Dropping the db name (if it isn't from global temp db) is performed in `isTempView`.

Reply:
>  I pass all parts in identifier

But you did not... The catalog name is missing, so you may mistakenly treat a table as temp view, e.g. `cat1.t1` if `t1` is the name of a temp view.

Reply:
Thanks for explaining. I see the failing case when catalog "exists" for the head of identifier; let me fix it immediately.